contrib/babelfishpg_tsql/src/pltsql_ruleutils.c

```diff
@@ -432,52 +434,53 @@ tsql_get_functiondef(PG_FUNCTION_ARGS)
 		appendStringInfoString(&buf, "(");
 	
 	/* we will not pfree name because as we can see name = NameStr(proc->proname) 
-         * here we are not allocating extra space for name, we’re just using proc-> proname.
-         * also at the end, we’re releasing proctup (that will free proc->proname).  
-         */
-	pfree(nsp);
-	if (nnsp)
-		pfree(nnsp);
-        
+	 * here we are not allocating extra space for name, we’re just using proc-> proname.
+	 * also at the end, we’re releasing proctup (that will free proc->proname).  
+	 */
+
 	tmp = SysCacheGetAttr(PROCOID, proctup, Anum_pg_proc_probin, &isnull);
-        number_args = proc->pronargs;
-        if(isfunction) number_args++;
+
+	if (!isnull)
+		probin_c = TextDatumGetCString(tmp);
+	if(!probin_c || probin_c[0] != '{')
+		PG_RETURN_NULL();
+	
+	number_args = proc->pronargs;
+	if(isfunction) number_args++;
```

Comment:
please add a linebreak after the condition

Reply:
done